src/strict_fstream.hpp

```diff
@@ -16,42 +15,52 @@
  */
 namespace strict_fstream
 {
-/// Overload of error-reporting function, to enable use with VS (1)
-/// and POSIX signature found in MUSL on Alpine (2)
-/// Ref 1: http://stackoverflow.com/a/901316/717706
-/// Ref 2: http://stackoverflow.com/a/41956165
-static inline char* strerror_r_compat(int result, char* buffer, int err)
-{
-    if (result)
-    {
-        sprintf(buffer, "Unknown error: %d", err);
-    }
-    return buffer;
-}
 
-static inline char* strerror_r_compat(char* result, char*, int)
+// Workaround for broken musl implementation
+// Since musl insists that they are perfectly compatible, ironically enough,
+// they don't have a __musl__ or similar. But __NEED_size_t is defined in their
+// relevant header (and not in working implementations), so we can use that.
+#ifdef __NEED_size_t
+#define BROKEN_GNU_STRERROR_R
+#warning "Working around broken strerror_r() implementation in musl, remove when musl is fixed"
+#endif
+
+// Non-gnu variants of strerror_* don't necessarily null-terminate if
+// truncating, so we have to do things manually.
+inline std::string &trim_to_null(std::string &buff)
 {
-    return result;
+    std::string::size_type pos = buff.find('\0');
+    if (pos == std::string::npos) {
+        buff += " [...]"; // it has been truncated
+    } else {
+        buff.resize(pos);
+    }
+    return buff;
 }
 
+/// Overload of error-reporting function, to enable use with VS.
+/// Ref: http://stackoverflow.com/a/901316/717706
```

Comment:
Maybe re-add current master links.

/// Overload of error-reporting function, to enable use with VS (1)
/// and POSIX signature found in MUSL on Alpine (2)
/// Ref 1: http://stackoverflow.com/a/901316/717706
/// Ref 2: http://stackoverflow.com/a/41956165

Reply:
It's a different approach to the MUSL issue than #25. Using a def in one of their headers, because they do not provide `__MUSL__`.

Reply:
I added a link to a discussion on the (official) musl mailing list instead of the second link.